### PR TITLE
Add query source to DB spans

### PR DESCRIPTION
**Pull request by antonpirker**
Adding OTel compatible information to database spans that show the code location of the query.
Refs https://github.com/getsentry/team-sdks/issues/40

This has to be 
- [x] "opt in" and 
- [x] only add the location where the query takes longer than `100ms` to run
- [x] working with SQLAlchemy
- [x] working with Django
- [x] working with asyncpg

Documentation PR: https://github.com/getsentry/sentry-docs/pull/8558

In the product this will then be displayed with something like this:

query-source

### CODE REVIEW

**Comment by AbhiPrasad** [comment]
Is it possible to tell what the overhead of this is going to be?

**Comment by antonpirker** [comment]
> Is it possible to tell what the overhead of this is going to be?

Hard to tell. I am "only" walking the stack until i find the right frame and then add the data to the span. But maybe we should do some load testing.

(it is opt-in, so we will not break anything out of the box)

**Comment by antonpirker** [comment]
@AbhiPrasad  Did some load testing on an FastAPI example app running 50 concurrent users doing around 85 requests per second:

Without attaching the query source:

without-query-source-50users-fastapi

With attaching the query source:

with-query-source-50users-fastapi

The only difference I can see is the 99percentile is a bit slower with attaching query sources (but those are probably some outliers, the the max column)

So I think we are good here.

**Comment by antonpirker** [comment]
And here the event as JSON: [python-with-query-source.json](https://github.com/getsentry/sentry-python/files/13440013/python-with-query-source.json)

**Comment by antonpirker** [comment]
One thing to mention: When finding the right frame to attach as query source I am omitting external sources (stuff that is in `site-packages` and `dist-packages`) 

This works well if someone has for example Django and does queries using Djangos ORM because the queries are done in the users code. But if they have Django and DjangoRestFramework the place in the users code that defines the query does not show up in the stack trace and I can not add the query source. 

Which is a bummer, but I think kind of ok, because if they have DjangoRestFramework the transaction name will lead them to their serializer and then they know where to look for the query/queries.

**Review by sentrivana — APPROVED** [review]
LGTM, left some suggestions

File: sentry_sdk/consts.py [hunk 1]
```diff
@@ -55,6 +55,8 @@
 DEFAULT_MAX_BREADCRUMBS = 100
 MATCH_ALL = r".*"
 
+DB_SPAN_DURATION_THRESHOLD_MS = 100
```
[the remark this answers is not in the document]
Comment (gggritso) [reply sentry_sdk/consts.py:58]: That would be great from the product side! People will definitely request this. I didn't make this prominent enough in the project proposal, sorry

File: sentry_sdk/tracing_utils.py [hunk 2]
```diff
@@ -162,6 +166,99 @@ def maybe_create_breadcrumbs_from_span(hub, span):
...
+        if code_function is not None:
+            span.set_data(SPANDATA.CODE_FUNCTION, frame.f_code.co_name)
+
+
+def add_additional_span_data(hub, span):
+    # type: (sentry_sdk.Hub, sentry_sdk.tracing.Span) -> None
+    """
+    Adds additional data to the span
+    """
+    if span.op == OP.DB:
```
[the remark this answers is not in the document]
Comment (antonpirker) [reply sentry_sdk/tracing_utils.py:257]: I have now moved the checks inot `add_query_source`. This makes this function cleaner.
I want to have the if here so it signals "you can add an elif for other span.op to add additional data"

**Pull request merged**